lib/block_view_lockups.go

```diff
@@ -1404,6 +2196,10 @@ func (bav *UtxoView) _connectCoinLockupTransfer(
 	// Fetch PKIDs for the recipient, sender, and profile.
 	senderPKIDEntry := bav.GetPKIDForPublicKey(txn.PublicKey)
 	senderPKID := senderPKIDEntry.PKID
+	if txMeta.RecipientPublicKey.IsZeroPublicKey() {
+		return 0, 0, nil,
+			errors.Wrap(RuleErrorCoinLockupTransferToZeroPublicKey, "_connectCoinLockupTransfer")
+	}
```

Comment:
Add a sanity check to this function as follows:
1. Snapshot the sender's balance at the beginning.
2. Toward the end, compute lockedBalanceDelta = sum(prevLockedBlanaceEntries) - sum(newLockedBalanceEntries).
3. Verify that newBalance + lockedBalanceDelta <= oldBalance

Reply:
Extra credit: Add a similar sanity-check to the lockup transfer function as well.